src/platform/BUILD.gn

```diff
@@ -47,6 +47,9 @@ if (chip_device_platform != "none") {
 
     # Enable including the additional data in the advertisement packets
     chip_enable_additional_data_advertising = true
+
+    # Enable commisionable discovery mode
+    chip_enable_commisionable_discovery = true
```

Comment:
@chrisdecenzo - I still have doubts about why we would want to have this as it complicates the build system. Why is just not having the application call Adverise::(CommisionableParams) sufficient?

Reply:
Optionality does not imply a compile flag: vendorid and productid advertisement is optional so we have made the data structure say Optional<uint16_t> and did not add a flag. My default thought is that this is the same: we expose a call (like we had setVendorId) and if the application never calls it, it is not used and that is fine.

Trying to understand why this would be different, since different involves more code complexity (build flags and code `#ifdefs`)